homeassistant/components/srp_energy/config_flow.py

```diff
@@ -1,66 +1,93 @@
 """Config flow for SRP Energy."""
-import logging
+from __future__ import annotations
+
+from typing import Any
 
 from srpenergy.client import SrpEnergyClient
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.const import CONF_ID, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
 
-from .const import CONF_IS_TOU, DEFAULT_NAME, SRP_ENERGY_DOMAIN
-
-_LOGGER = logging.getLogger(__name__)
+from .const import CONF_IS_TOU, DOMAIN, LOGGER
 
 
-class ConfigFlow(config_entries.ConfigFlow, domain=SRP_ENERGY_DOMAIN):
-    """Handle a config flow for SRP Energy."""
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
 
-    VERSION = 1
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    LOGGER.debug("Validate User Input with client")
+    srp_client = SrpEnergyClient(
+        data[CONF_ID],
+        data[CONF_USERNAME],
+        data[CONF_PASSWORD],
+    )
 
-    config = {
-        vol.Required(CONF_ID): str,
-        vol.Required(CONF_USERNAME): str,
-        vol.Required(CONF_PASSWORD): str,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
-        vol.Optional(CONF_IS_TOU, default=False): bool,
-    }
+    is_valid = await hass.async_add_executor_job(srp_client.validate)
 
-    async def async_step_user(self, user_input=None):
-        """Handle a flow initialized by the user."""
-        errors = {}
+    LOGGER.debug("Is valid User Input with client: %s", is_valid)
```

Comment:
This sentence doesn't make sense. Please use correct English including capitalization when logging.

Reply:
Will fix